### ISSUE TITLE
Cache sub-component instance when rendering

### ISSUE THREAD

**Issue by mihaimd** [lit/lit#4754]
### Should this be an RFC?

- [X] This is not a substantial change

### Which package is this a feature request for?

Task (@lit/task)

### Description

When you render a sub-component instance in the complete function of a Task the sub-component is being re-instantiated every time the Task re-runs. This is not great because thus the sub-component will lose state every time. 

It would be great if the sub-component instance would just get the updated bound data and the DOM changes would happen only where that data is used without re-rendering the whole component again. 

Right now if you don't have the case where you would run the Task only once and render a sub-component the Task feature is not very useful for using dumb sub-components that need data updates in response to some user interaction.

Thank you! 

Code example:
`import { LitElement, css, html } from 'lit'
import { MyComponent } from './my-component';
import { Task } from '@lit/task';

export class MyElement extends LitElement {
  static get properties() {
    return {
      count: { type: Number }
    }
  }

  constructor() {
    super()
    this.count = 0
  }

  _apiTask = new Task(
    this,
    ([count]) => count,
    () => [this.count]
  );

    render() {
      return html`
            ${this._apiTask.render({
          pending: () => html`Loading user info...`,
          complete: (count) => html`<my-component .data="${count}"></my-component>`,
        })}
      `
    }

    _onClick() {
      this.count++
    }
  }
}

window.customElements.define('my-element', MyElement)`

### Alternatives and Workarounds

I guess a workaround would be to keep that state outside of the child component in the parent component and feed it via bound reactive properties every time.

**Comment by sorvell** [comment]
This is pretty easy to do using the `cache` directive. Here's an [example](https://lit.dev/playground/#gist=baab3d0f821191e798e54b3f50e2447a).

`cache(this._apiTask.render({...})`

In general, it's not clear that you'd aways want this behavior, and it'd also be a breaking change. So we probably wouldn't just change the base behavior, but probably for the majority of cases, yes, this makes sense and you can use `cache` to do it.

**Comment by justinfagnani** [comment]
I think `cache(this._apiTask.render({...})`is the right solution here. We strive for features that compose well together, and adding caching on top of task's render helper does that well and is the same pattern you'd use for any other render helper.

I think we can close this as working. @mihaimd if this doesn't work for you, let us know.

**Comment by mihaimd** [comment]
Thank you for replying, I went with the other solution to render my component outside the task and just use the task to get the data, but using the cache makes total sense. I will refactor in the future.

All the best,
Mihai
